tests/apimatic_core/type_combinator_tests/test_one_of.py

```diff
@@ -0,0 +1,793 @@
+             UnionTypeContext().dict(True).array(True).array_of_dict(True), False,
+             None),
+
+            # dictionary of array cases
+            ({'key0': ['abc', 'def'], 'key1': ['ghi', 'jkl']},
+             [LeafType(int, UnionTypeContext().dict(True).array(True)),
+              LeafType(str, UnionTypeContext().dict(True).array(True))],
+             UnionTypeContext(), True, {'key0': ['abc', 'def'], 'key1': ['ghi', 'jkl']}),
+            ({'key0': [100, 200], 'key1': [300, 400]},
+             [LeafType(int, UnionTypeContext().dict(True).array(True)),
+              LeafType(str, UnionTypeContext().dict(True).array(True))],
+             UnionTypeContext(), True, {'key0': [100, 200], 'key1': [300, 400]}),
+            ({'key0': ['abc', 200], 'key1': ['def', 400]},
+             [LeafType(int, UnionTypeContext().dict(True).array(True)),
+              LeafType(str, UnionTypeContext().dict(True).array(True))],
+             UnionTypeContext(), False, None),
+            ({'key0': [100, 200], 'key1': ['abc', 'def']},
+             [LeafType(int, UnionTypeContext().dict(True).array(True)),
+              LeafType(str, UnionTypeContext().dict(True).array(True))],
+             UnionTypeContext(), False, None),
+            ([{'key0': [100, 200]}, {'key1': ['abc', 'def']}],
+             [LeafType(int, UnionTypeContext().dict(True).array(True)),
+              LeafType(str, UnionTypeContext().dict(True).array(True))],
+             UnionTypeContext(), False, None),
+
+            # Outer dictionary of array cases
+            ({'key0': ['abc', 'def'], 'key1': ['ghi', 'jkl']},
+             [LeafType(int, UnionTypeContext()), LeafType(str, UnionTypeContext())],
+             UnionTypeContext().dict(True).array(True), True, {'key0': ['abc', 'def'], 'key1': ['ghi', 'jkl']}),
+            ({'key0': [100, 200], 'key1': [300, 400]},
+             [LeafType(int, UnionTypeContext()), LeafType(str, UnionTypeContext())],
+             UnionTypeContext().dict(True).array(True), True, {'key0': [100, 200], 'key1': [300, 400]}),
+            ({'key0': ['abc', 200], 'key1': ['def', 400]},
+             [LeafType(int, UnionTypeContext()), LeafType(str, UnionTypeContext())],
+             UnionTypeContext().dict(True).array(True), True, {'key0': ['abc', 200], 'key1': ['def', 400]}),
+            ({'key0': [100, 200], 'key1': ['abc', 'def']},
+             [LeafType(int, UnionTypeContext()), LeafType(str, UnionTypeContext())],
+             UnionTypeContext().dict(True).array(True), True, {'key0': [100, 200], 'key1': ['abc', 'def']}),
+            ([{'key0': [100, 200]}, {'key1': ['abc', 'def']}],
+             [LeafType(int, UnionTypeContext()), LeafType(str, UnionTypeContext())],
+             UnionTypeContext().dict(True).array(True), False, None),
+            ([[100, 200], None],
+             [OneOf([LeafType(str, UnionTypeContext()), LeafType(bool, UnionTypeContext())],
```

Comment:
only this test input have `OneOf` specified, is this a test for inner cases?

Reply:
Yes, other than this case, there are more cases added for nested unit test cases.